build.sh

```diff
@@ -414,17 +424,32 @@ fail_handler() {
     exit 1
 }
 
+get_artifacts() {
+    if [ "$os" = "win" ]; then
+        ssp_artifacts="libssp-0.dll"
+        winpthreads_artifacts="libwinpthread-1.dll"
+        openslide_artifacts="libopenslide-0.dll openslide-quickhash1sum.exe openslide-show-properties.exe openslide-write-png.exe"
+        openslide_java_artifacts="openslide-jni.dll openslide.jar"
+    elif [ "$os" = "linux" ]; then
+        ssp_artifacts="libssp.so"
+        winpthreads_artifacts=""
+        openslide_artifacts="libopenslide.so.0 openslide-quickhash1sum openslide-show-properties openslide-write-png"
```

Comment:
Hmm.  Under our existing policy, if we statically linked glibc, we'd need to ship the corresponding source for it, which means in practice that we'd have to build it ourselves.  We probably don't want to do that.  The alternative is to create a second container image for building the Linux binaries, based on the _oldest_ Linux distro version we want to support.

Reply:
Unsure if this is what you're after, but `objdump -x` gives

```
Dynamic Section:
  NEEDED               libm.so.6
  NEEDED               libpcre2-8.so.0
  NEEDED               libmount.so.1
  NEEDED               libselinux.so.1
  NEEDED               libc.so.6
  NEEDED               ld-linux-x86-64.so.2
```

Reply:
Yup, that's what I'm after.  `libmount` is from util-linux, so Linux systems will probably have it, and `libm`/`libc`/`ld-linux` are standard libraries.  But the build is currently assuming that the system has `libselinux` (which some distros won't ship) and a particular version of PCRE2.

Reply:
Can remove selinux by setting `'selinux=disabled'` for the `glib` subproject, but I'm not sure about `libpcre`. Especially given `'default_library=static'` is already set globally

Reply:
Any ideas for pcre? I've not figured out why it's grabbing the shared library and no amount of encouraging static libraries seems to change that.

Reply:
Unless I'm mistaken, `libpcre2-8` isn't a particular version of PCRE2, it's just the 8-bit library of PCRE2, so `libpcre-dev` in any package manager will include it. I don't really see an issue with linking it dynamically https://packages.debian.org/sid/libpcre2-8-0

Sorry to bump this again after a long wait, but we're trying to get ready for the next QuPath release. Would like to get this merged if possible or at least a version of the dependencies yous would be happy to get redistributed

Reply:
I'll probably need to spend a lot of time with this PR before merging it, and unfortunately that won't happen in the next few weeks.  For the next QuPath release I'd recommend either skipping these changes or shipping from your own openslide-winbuild fork, as you prefer.

For dependency versions, OpenSlide doesn't have strong preferences, except that pixman 0.38.x should be avoided.  We already need to support whatever dependencies are shipped with the various Linux distros.

Re dynamically linking libpcre: isn't the goal is to avoid any external dependencies, though?

Reply:
Yeah fair enough, I think there are a lot of issues remaining with the build anyways - still seeing a lot of hardcoded paths that are less than ideal. Will revisit this later, perhaps with some stronger ideas for a solution if we find something reasonable for our use case.